lib/enkf/value_export.cpp

```diff
@@ -46,86 +48,147 @@ static void backup_if_existing(const char * filename) {
 
 
 value_export_type * value_export_alloc(const char * directory, const char * base_name) {
-  value_export_type * value = (value_export_type *)util_malloc( sizeof * value );
+  value_export_type * value = new value_export_type;
   UTIL_TYPE_ID_INIT( value , VALUE_EXPORT_TYPE_ID );
-  value->directory = util_alloc_string_copy( directory );
-  value->base_name = util_alloc_string_copy( base_name );
 
-  value->keys = stringlist_alloc_new( );
-  value->values = double_vector_alloc(0,0);
+  if(directory == NULL)
+    value->directory = "";
+  else
+    value->directory = directory;
+  if(base_name == NULL)
+    value->base_name = "";
```

Comment:
We have no intention of handling `base_name == NULL` - that is exception material!

Reply:
Certain tests and calls gives NULL as directory. NULL handling was added to keep original code working. This was acceptable in the old version as it was stored as a pointer. I can change the function calls to use string if that is desired.